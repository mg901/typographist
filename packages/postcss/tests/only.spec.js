const run = require('./run');
const { userConfig } = require('../../../mocks');

describe('@only', () => {
  it('should replace @only(desktop) with @media (min-width: 48em) and (max-width: 61.99875em)', () => {
    const source = `
          .test {
            @only (desktop) {
              background-color: rebeccapurple;
            }
          }`;
    const compiled = `
          @media (min-width: 62em) and (max-width: 74.99875em) {
    .test {
        background-color: rebeccapurple;
    }
}`;
    return run(source, compiled, userConfig);
  });
});

describe('@only atrule with orientation', () => {
  it('should replace @only(desktop):landscape with @media (min-width: 48em) and (max-width: 61.99875em) and (orientation: landscape)', () => {
    const source = `
          .test {
            @only (desktop):landscape {
              background-color: rebeccapurple;
            }
          }`;
    const compiled = `
          @media (min-width: 62em) and (max-width: 74.99875em) and (orientation: landscape) {
    .test {
        background-color: rebeccapurple;
    }
}`;
    return run(source, compiled, userConfig);
  });
});
