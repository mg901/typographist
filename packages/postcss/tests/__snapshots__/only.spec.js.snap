// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@only with breakpoints name 1`] = `
@media (min-width: 62em) and (max-width: 74.99875em) {
  .test {
    background-color: rebeccapurple;
  }
}
`;

exports[`@only with em 1`] = `
@media (min-width: 40em) and (max-width: 39.99875em) {
  .test {
    background-color: gold;
  }
}
`;

exports[`@only with orientation \`landscape\` 1`] = `
@media (min-width: 62em) and (max-width: 74.99875em) and (orientation: landscape) {
  .test {
    background-color: gold;
  }
}
`;

exports[`@only with orientation \`portrait\` 1`] = `
@media (min-width: 62em) and (max-width: 74.99875em) and (orientation: portrait) {
  .test {
    background-color: gold;
  }
}
`;

exports[`@only with pixels 1`] = `
@media (min-width: 62.5em) and (max-width: 62.49875em) {
  .test {
    background-color: gold;
  }
}
`;
